checksec/__main__.py

```diff
@@ -27,15 +28,15 @@
 from .utils import lief_set_logging
 
 
-def walk_filepath_list(filepath_list: List[Path], recursive: bool = False) -> Iterator[Path]:
+def walk_filepath_list(filepath_list: List[Path], recursive: bool = False, ignore_symlinks: bool = False) -> Iterator[Path]:
     for path in filepath_list:
         if path.is_dir() and not path.is_symlink():
             if recursive:
                 for f in os.scandir(path):
-                    yield from walk_filepath_list([Path(f)], recursive)
+                    yield from walk_filepath_list([Path(f)], recursive, ignore_symlinks)
             else:
                 yield from (Path(f) for f in os.scandir(path))
```

Comment:
The issue is that this line will yield any path, and not filter on the symlink as expected.
for example, if you have some symlinks in `/usr/lib` (`/usr/lib/cpp -> `/etc/alternatives/cpp`),
running `checksec.py -i /usr/lib` will still yield `/usr/lib/cpp` because it goes through that `yield from`

Can you find a way to rework that function ?

Reply:
This is a fix, I duplicated the condition:
~~~python3
def walk_filepath_list(filepath_list: List[Path], recursive: bool = False, ignore_symlinks: bool = False) -> Iterator[Path]:
    for path in filepath_list:
        if path.is_dir() and not path.is_symlink():
            if recursive:
                for f in os.scandir(path):
                    yield from walk_filepath_list([Path(f)], recursive, ignore_symlinks)
            else:
                yield from (Path(f) for f in os.scandir(path) if f.is_file()
                        and (not ignore_symlinks or not f.is_symlink()))
        elif path.is_file() and (not ignore_symlinks or not path.is_symlink()):
            yield path
~~~

but i don't like duplication :confused:

Reply:
hope switching to os.walk() is fine for you, seemed a bit simpler to me ;-)